File: class_6/class_6/app/src/main/java/com/alejouribesanchez/class_6/Main7Activity.java
package com.alejouribesanchez.class_6;

import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;
import android.support.v4.graphics.drawable.DrawableCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;

public class Main7Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main7);

        EditText editUsername = (EditText) findViewById(R.id.username);

        editUsername.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                // ¿v tiene el foco?
                if(hasFocus){
                    ImageView iconoDescuento = (ImageView) findViewById(R.id.ic_account);
                    Drawable d = iconoDescuento.getDrawable();

                    d = DrawableCompat.wrap(d);
                    DrawableCompat.setTint(d,
                            ContextCompat.getColor(Main7Activity.this, R.color.colorAccent));

                }
            }
        });
    }

}

File: class_6/class_6/app/src/main/java/com/alejouribesanchez/class_6/Main7Activity.kt
package com.alejouribesanchez.class_6

import android.graphics.drawable.Drawable
import android.support.v4.content.ContextCompat
import android.support.v4.graphics.drawable.DrawableCompat
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.ImageView

class Main7Activity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main7)

        val editUsername = findViewById(R.id.username) as EditText

        editUsername.onFocusChangeListener = View.OnFocusChangeListener { v, hasFocus ->
            // ¿v tiene el foco?
            if (hasFocus) {
                val iconoDescuento = findViewById(R.id.ic_account) as ImageView
                var d = iconoDescuento.drawable

                d = DrawableCompat.wrap(d)
                DrawableCompat.setTint(d,
                        ContextCompat.getColor(this@Main7Activity, R.color.colorAccent))

            }
        }
    }

}
